client.go

```diff
@@ -93,28 +97,95 @@ func NewClient(cfg Config) (c *Client, err error) {
 	upgradetypes.RegisterInterfaces(interfaceRegistry)
 	feegranttypes.RegisterInterfaces(interfaceRegistry)
 
+	tmClient, err := client.NewClientFromNode(cfg.RPCHost)
+	if err != nil {
+		return nil, fmt.Errorf("NewClientFromNode error: %w", err)
+	}
+
 	conn, err := grpc.NewClient(cfg.GRPCHost, grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")))
 	if err != nil {
 		return nil, fmt.Errorf("grpc.Dial: %s %s", cfg.GRPCHost, err)
 	}
 
-	cancelCtx, cancelFn := context.WithCancel(context.Background())
+	cosmosKeyring := keyring.NewInMemory()
+	path := hd.CreateHDPath(118, 0, 0).String()
+
+	senderInfo, err := cosmosKeyring.NewAccount(cfg.SignerName, cfg.SignerMnemonic, "", path, hd.Secp256k1)
+	if err != nil {
+		return nil, fmt.Errorf("cosmosKeyring.NewAccount error: %w", err)
+	}
+
+	marshaller := codec.NewProtoCodec(interfaceRegistry)
+	txConfig := tx.NewTxConfig(marshaller, []signing.SignMode{signing.SignMode_SIGN_MODE_DIRECT})
+
+	clientCtx := client.Context{
+		ChainID:       cfg.ChainID,
+		BroadcastMode: flags.BroadcastAsync,
+		TxConfig:      txConfig,
+	}.WithKeyring(cosmosKeyring).WithFromAddress(senderInfo.GetAddress()).
+		WithFromName(senderInfo.GetName()).WithFrom(senderInfo.GetName()).
+		WithAccountRetriever(authtypes.AccountRetriever{}).WithClient(tmClient).
+		WithInterfaceRegistry(interfaceRegistry)
+
+	txFactory := new(txf.Factory).
+		WithKeybase(clientCtx.Keyring).
+		WithTxConfig(clientCtx.TxConfig).
+		WithAccountRetriever(clientCtx.AccountRetriever).
+		WithSimulateAndExecute(true).
+		WithGasAdjustment(1.1).
+		WithChainID(clientCtx.ChainID).
+		WithSignMode(signing.SignMode_SIGN_MODE_DIRECT).
+		WithGasPrices(DefaultGasPriceWithDenom)
+
+	sgn := &sign{
```

Comment:
should it be a separate structure?

Reply:
I think if we will keep it separate - it would be easier to read